scenarios/smart_trailer/digital_twin_providers/trailer_properties_provider/src/trailer_properties_provider_impl.rs

```diff
@@ -0,0 +1,272 @@
+};
+
+use digital_twin_model::{trailer_v1, Metadata};
+use log::{debug, info, warn};
+use paho_mqtt as mqtt;
+use parking_lot::RwLock;
+use serde_derive::{Deserialize, Serialize};
+use strum_macros::{Display, EnumString};
+use tokio::sync::{mpsc, watch};
+use tokio::task::JoinHandle;
+use tokio::time::{sleep, Duration};
+use tonic::{Request, Response, Status};
+
+const MQTT_CLIENT_ID: &str = "trailer-properties-publisher";
+const FREQUENCY_MS: &str = "frequency_ms";
+
+#[derive(Debug, Serialize, Deserialize)]
+struct TrailerWeightProperty {
+    #[serde(rename = "TrailerWeight")]
+    trailer_weight: trailer_v1::trailer::trailer_weight::TYPE,
+    #[serde(rename = "$metadata")]
+    metadata: Metadata,
+}
+
+/// Actions that are returned from the Pub Sub Service.
+#[derive(Clone, EnumString, Eq, Display, Debug, PartialEq)]
+pub enum ProviderAction {
+    #[strum(serialize = "PUBLISH")]
+    Publish,
+
+    #[strum(serialize = "STOP_PUBLISH")]
+    StopPublish,
+}
+
+#[derive(Debug)]
+pub struct TopicInfo {
+    topic: String,
+    stop_channel: mpsc::Sender<bool>,
+}
+
+#[derive(Debug)]
+pub struct TrailerPropertiesProviderImpl {
+    pub data_stream: watch::Receiver<i32>,
+    pub min_interval_ms: u64,
+    entity_map: Arc<RwLock<HashMap<String, Vec<TopicInfo>>>>,
+}
+
+/// Create the JSON for the trailer weight property.
+///
+/// # Arguments
+/// * `trailer_weight` - The trailer weight value.
+fn create_property_json(trailer_weight: i32) -> String {
+    let metadata = Metadata {
+        model: trailer_v1::trailer::trailer_weight::ID.to_string(),
+    };
+
+    let property: TrailerWeightProperty = TrailerWeightProperty {
+        trailer_weight,
+        metadata,
+    };
+
+    serde_json::to_string(&property).unwrap()
+}
+
+/// Publish a message to a MQTT broker located.
+///
+/// # Arguments
+/// `broker_uri` - The MQTT broker's URI.
+/// `topic` - The topic to publish to.
+/// `content` - The message to publish.
+fn publish_message(broker_uri: &str, topic: &str, content: &str) -> Result<(), String> {
+    let create_opts = mqtt::CreateOptionsBuilder::new()
+        .server_uri(broker_uri)
+        .client_id(MQTT_CLIENT_ID.to_string())
+        .finalize();
+
+    let client = mqtt::Client::new(create_opts)
+        .map_err(|err| format!("Failed to create the client due to '{err:?}'"))?;
+
+    let conn_opts = mqtt::ConnectOptionsBuilder::new()
+        .keep_alive_interval(Duration::from_secs(30))
+        .clean_session(true)
+        .finalize();
+
+    let _connect_response = client
+        .connect(conn_opts)
+        .map_err(|err| format!("Failed to connect due to '{err:?}"));
+
+    let msg = mqtt::Message::new(topic, content, mqtt::types::QOS_1);
+    if let Err(err) = client.publish(msg) {
+        return Err(format!("Failed to publish message due to '{err:?}"));
+    }
+
+    if let Err(err) = client.disconnect(None) {
+        warn!("Failed to disconnect from topic '{topic}' on broker {broker_uri} due to {err:?}");
+    }
```

Comment:
Elsewhere (`scenarios/smart_trailer/applications/smart_trailer_application/src/main.rs` line 148) this has been treated as an error. Should it be an error here too? if so this is the suggested change:

```suggestion
    client.disconnect(None)
        .map_err(|err| format!("Failed to disconnect from topic '{topic}' on broker {broker_uri} due to {err:?}"))?;
```